Fix the HUD to work with languages other than JRuby

snim2 opened the pull request:
- Fixes an old bug in cPython profiling
- Greys out HUD buttons for languages that cannot provide HUD information 
  [see this video for an example](https://youtu.be/zFFjqm90zDc)
- Displays onle the tooltips etc. that the user has requested via the HUD. 

e.g. user has selected callgraph information:

![hud-with-tooltip](https://cloud.githubusercontent.com/assets/97674/18203448/075b90b8-710f-11e6-85db-bc5be6bd2828.png)

Here the user has selected "No HUD" so the tooltip is not displayed:

![no-hud-no-tooltip](https://cloud.githubusercontent.com/assets/97674/18203456/17830b24-710f-11e6-8483-0205f3276a58.png)

Fixes #107

Commit:
CPython heat map works with recursive calls.

Commit:
Remove unused code. Truffle languages (SimpleLanguage included) no longer have coverage counters.

Commit:
HUD greyed out when no profile information is available.

ltratt commented:
Maybe we should do a PR that's just whitespace removal? I must admit, it is a bit distracting to have it in every PR. [I agree, though, that getting rid of it is generally a good thing.]

snim2 commented:
Hopefully this is the last of it. Do you want me to cherry-pick that commit into another PR?

ltratt commented:
Ah, if it's the last one, don't worry about it.

Commit:
Display tooltips for correct HUD selection.
This commit separates out annotations that tell Eco how to render visualisation
information (e.g. heatmap, railroad diagram, etc.), from annotations that tell
Eco which HUD option the information matches. This means that where Eco has
(for example) different tooltips for JRuby types and the JRuby callgraph,
the user can be give only the tooltips they have requested via the HUD.

snim2 commented:
This is ready for review now.

ltratt commented:
Looks good. Do you want to squash and then I can merge?

snim2 commented:
OK, squashed!

ltratt commented:
Will merge once tests have completed.

snim2 commented:
Thanks
